morpheus/_lib/src/messages/control.cpp

```diff
@@ -41,12 +42,30 @@ ControlMessage::ControlMessage(const nlohmann::json& _config) :
   m_tasks({})
 {
     config(_config);
+    // Register cleanup for this instance
+    auto cleanup = [this]() {
+        py::gil_scoped_acquire acquire;  // Acquire the GIL
+        this->m_tensors.reset();         // Clear the shared_ptr
+    };
```

Comment:
We want to be handling this a different way and not with `atexit`

Reply:
Yeah, this is removed. It wasn't related to the behavior I was observing.